university/src/university/AbstractFactory.java

```diff
@@ -0,0 +1,5 @@
+package university;
+
+public interface AbstractFactory<T> {
+    T create(Class<? extends T> clazz);
```

Comment:
Fancy indeed - but why is it here if it is not implemented / used at all? You could have some time to create a concrete factory implementing this interface to create different `Enrollable` implementations and subsequently use that factory in the main method to create different Students or Researchers - this way, you could understand how factories work and what there advantages are.